### PR TITLE
Add range op

### PR DESCRIPTION
**Pull request by kahmed10**
#481

### CODE REVIEW

File: src/onnx/onnx.cpp [hunk 1]
```diff
@@ -1989,6 +1990,47 @@ struct onnx_parser
...
+            assert(delta_val != 0);
+
+            size_t num_elements = static_cast<size_t>(
+                ceil(static_cast<double>(limit_val - start_val) / static_cast<double>(delta_val)));
+
+            assert(num_elements > 0);
+
+            using type = decltype(start_val);
+
+            std::vector<type> range_vals;
```
Comment (pfultz2) [inline src/onnx/onnx.cpp:2022]: You should preallocate the vector.

File: src/onnx/onnx.cpp [hunk 2]
```diff
@@ -1989,6 +1990,47 @@ struct onnx_parser
...
+
+            for(int64_t i = 0; i < num_elements; i++)
+            {
+                range_vals.push_back(start_val + static_cast<type>(i) * delta_val);
+            }
```
Comment (pfultz2) [inline src/onnx/onnx.cpp:2027]: You could write this using `std::generate`:

```cpp
std::generate(range_vals.begin(), range_vals.end(), [&]() { 
    auto result = start_val;
    start_val += delta_val;
    return result;
});
```

Comment (kahmed10) [reply src/onnx/onnx.cpp]: fixed

File: src/onnx/onnx.cpp (hunk 1 above)
Comment (kahmed10) [reply src/onnx/onnx.cpp]: fixed

File: src/onnx/onnx.cpp [hunk 3]
```diff
@@ -1989,6 +1990,47 @@ struct onnx_parser
...
+
+        auto start_arg = args[0]->eval();
+        check_arg_empty(start_arg, "PARSE_RANGE: start arg dynamic shape is not supported");
+        auto limit_arg = args[1]->eval();
+        check_arg_empty(limit_arg, "PARSE_RANGE: limit arg dynamic shape is not supported");
+        auto delta_arg = args[2]->eval();
+        check_arg_empty(delta_arg, "PARSE_RANGE: delta arg dynamic shape is not supported");
+
+        assert(args[0]->get_shape().elements() == args[1]->get_shape().elements() ==
+               args[2]->get_shape().elements() == 1);
```
Comment (pfultz2) [inline src/onnx/onnx.cpp:2005]: Actually, this does not do what is expected. `args[0]->get_shape().elements() == args[1]->get_shape().elements()` will return a boolean. It should be `args[0]->get_shape().elements() == 1 && args[1]->get_shape().elements() == 1 && args[2]->get_shape().elements() == 1`.

Comment (kahmed10) [reply src/onnx/onnx.cpp:2005]: fixed

**Pull request merged**